src/main/java/com/mmm/his/cer/utility/farser/ast/parser/ExpressionResult.java

```diff
@@ -3,32 +3,32 @@
 /**
  * Class that contains all information about a expression match from {@link DescentParser}.
  *
- * @param <T> the type of context
+ * @param <C> the type of context which was used for the terminal nodes to evaluate the AST
  * @author Mike Funaro
  */
-public class ExpressionResult<T> {
+public class ExpressionResult<C, R> {
 
-  private final boolean matched;
-  private final T context;
+  private final R result;
+  private final C context;
 
   /**
    * Ctor.
    *
-   * @param matched boolean true if the expression evaluated to true, false otherwise. When there is
+   * @param result boolean true if the expression evaluated to true, false otherwise. When there is
    *                true in the matched field the contents of the matches field will have the
    *                terminal objects that were matched.
    * @param context the context used in evaluation.
    */
-  public ExpressionResult(boolean matched, T context) {
-    this.matched = matched;
+  public ExpressionResult(R result, C context) {
+    this.result = result;
     this.context = context;
   }
 
-  public boolean isMatched() {
-    return matched;
+  public R getResult() {
```

Comment:
Breaking change, since also our existing implementations would call this I am guessing?

Reply:
Yes, true, I will bump the major version for this.

Reply:
Bumped to version 5.0.0